packages/dev-console-standalone/src/hooks/useDevConsoleInternal.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+  DevConsoleContext,
+  ExtensionPayload,
+  useDevConsole,
+} from '@shopify/ui-extensions-dev-console';
+import {useContext} from 'react';
+
+export function useDevConsoleInternal() {
+  const devConsole = useDevConsole();
+  const {host} = useContext(DevConsoleContext);
+  const {dispatch, update} = devConsole;
+
+  return {
+    ...devConsole,
+    host,
+
+    // update events
+    hide: (extensions: ExtensionPayload[]) =>
+      update({
+        extensions: extensions.map((extension) => ({
+          uuid: extension.uuid,
+          development: {hidden: true},
+        })),
+      }),
+    show: (extensions: ExtensionPayload[]) =>
+      update({
+        extensions: extensions.map((extension) => ({
+          uuid: extension.uuid,
+          development: {hidden: false},
+        })),
+      }),
+
+    // dispatch events
```

Comment:
I'm not sure if "dispatch" is a standard naming convention for events of this nature, but it doesn't seem to capture the meaning of these events. I'm not sure I have a better suggestion at the moment and feel free to disregard if it makes sense to everyone else. :)